pkg/urbit/vere/io/term.c

```diff
@@ -333,17 +335,48 @@ _term_it_write(u3_utty*  uty_u,
                uv_buf_t* buf_u,
                void*     ptr_v)
 {
-  uv_write_t* wri_u = c3_malloc(sizeof(*wri_u));
-  c3_w        ret_w;
+  //  work off a local copy of the buffer, in case we need
+  //  to manipulate the length/pointer
+  //
+  uv_buf_t     fub_u = { .base = buf_u->base, .len = buf_u->len };
+  uv_stream_t* han_u = (uv_stream_t*)&(uty_u->pop_u);
+  c3_i         ret_i;
 
-  wri_u->data = ptr_v;
+  //  try to write synchronously
+  //
+  while ( 1 ) {
+    ret_i = uv_try_write(han_u, &fub_u, 1);
```

Comment:
Under what conditions would this fail?

Reply:
There are a variety of error conditions for a bad call (bad parameters, invalid handle state, &c), there are failures in the underlying `write(2)` call that can bubble up, there are partial writes, and then there is `UV_EAGAIN`, which means that the stream is still connecting, or has asynchronous writes already queued. If we always use `uv_try_write()` first, and the stream is always already "connected" (by virtue of being a `uv_pipe_t`), I'm not sure that we'll ever end up with asynchronous writes.

Arguably, the terminal should just be completely synchronous, but that's a significantly larger change.